lib/libvarnishapi/vut.c

```diff
@@ -319,15 +323,18 @@ VUT_Fini(struct VUT **vutp)
 	CHECK_OBJ_NOTNULL(vut, VUT_MAGIC);
 	AN(vut->progname);
 
+	errno = 0;
+	if (pfh != NULL && VPF_Remove(pfh) != 0)
+		VUT_Error(vut, 1, "Cannot remove pid file %s: %s", vut->P_arg,
```

Comment:
`-P` is a global option, it isn't tied to a single VUT.

Reply:
Um, yes it is tied to the VUT object? It's a field in `struct VUT`? And all of the VUT apps in the Varnish distribution except for varnishhist hand off the `-P` arg to `VUT_Arg()`, which sets the field? (varnishhist uses -P for something else, and doesn't have an arg for a PID file.)